From 49c2e0c78166e1afdaaacdb3ceb241ed9aea035b Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 5 Apr 2022 11:38:43 +0200
Subject: [PATCH] Make APP_ID match real application ID

The application advertises itself as io.github.pwr_solaar.solaar through
its appdata file, so name the application this way too.

This fixes this warning in Flatpak:
Failed to register: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
---
 lib/solaar/cli/config.py  | 2 +-
 lib/solaar/ui/__init__.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/solaar/cli/config.py b/lib/solaar/cli/config.py
index 55e2165..0b165e3 100644
--- a/lib/solaar/cli/config.py
+++ b/lib/solaar/cli/config.py
@@ -187,7 +187,7 @@ def run(receivers, args, find_receiver, find_device):
         _print_setting(setting)
         return
 
-    APP_ID = 'io.github.pwr.solaar'
+    APP_ID = 'io.github.pwr_solaar.solaar'
     application = Gtk.Application.new(APP_ID, Gio.ApplicationFlags.HANDLES_COMMAND_LINE)
     application.register()
 
diff --git a/lib/solaar/ui/__init__.py b/lib/solaar/ui/__init__.py
index 4676fb6..7c05e6d 100644
--- a/lib/solaar/ui/__init__.py
+++ b/lib/solaar/ui/__init__.py
@@ -163,7 +163,7 @@ def _shutdown(app, shutdown_hook):
 def run_loop(startup_hook, shutdown_hook, use_tray, show_window):
     assert use_tray or show_window, 'need either tray or visible window'
     # from gi.repository.Gio import ApplicationFlags as _ApplicationFlags
-    APP_ID = 'io.github.pwr.solaar'
+    APP_ID = 'io.github.pwr_solaar.solaar'
     application = Gtk.Application.new(APP_ID, Gio.ApplicationFlags.HANDLES_COMMAND_LINE)
 
     application.connect('startup', lambda app, startup_hook: _startup(app, startup_hook, use_tray, show_window), startup_hook)
-- 
2.35.1

